gtsam/slam/dataset.h

```diff
@@ -218,8 +218,9 @@ typedef std::pair<size_t, size_t> SiftIndex;
 
 /// Define the structure for the 3D points
 struct SfmTrack {
-  SfmTrack(): p(0,0,0) {}
-  SfmTrack(const gtsam::Point3& pt) : p(pt) {}
+  SfmTrack(float r = 0, float g = 0, float b = 0): p(0,0,0), r(r), g(g), b(b) {}
+  SfmTrack(const gtsam::Point3& pt, float r = 0, float g = 0, float b = 0) : p(pt), r(r), g(g), b(b) {}
```

Comment:
Formatting please. Also, a good convention is to have struct/class variables be something like `r_` and then have the initializer list be `r_(r)`. This is consistent with Google C++ style which we follow.

Reply:
I like the style suggestion, but this would involve renaming all of the variables in the class, since none adhere to that guideline. @dellaert would you like to see a big change like that?

Reply:
I would vote for it since due to the previous lack of getters, it won't be an API change, so we should be able to maintain backwards compatibility.

Reply:
Actually, for *classes* the convention is underscore, for structs it is not (and should not).
It is an API change as these fields are public.

Reply:
Got it, thanks

Reply:
Ah you're right. My mistake.